src/serializer/functions.js

```diff
@@ -225,7 +226,29 @@ export class Functions {
     const globalThis = this.realm.$GlobalEnv.environmentRecord.WithBaseObject();
     let call = funcValue.$Call;
     invariant(call);
-    return call.bind(this, globalThis, []);
+    let numArgs = funcValue.getLength();
+    let args = [];
+    invariant(funcValue instanceof ECMAScriptSourceFunctionValue);
+    let params = funcValue.$FormalParameters;
+    if (numArgs && numArgs > 0 && params) {
+      for (let parameterId of params) {
+        if (t.isIdentifier(parameterId)) {
+          // Create a generic AbstractValue
```

Comment:
This is more than just a generic AbstractValue. It is effectively a call to global.__abstract expressed at the meta level. That probably needs a bit more explanation in the comment.

In fact, that is probably wrong, since a given name can only be used once per realm and this is too much to ask for when the name is a parameter name. Either qualify the name, or create a value like I do in checkResidualFunctions in prepack-standalone.js.